SimG4Core/Application/src/LowEnergyFastSimModel.cc

```diff
@@ -0,0 +1,62 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "SimG4Core/Application/interface/LowEnergyFastSimModel.h"
+
+#include "G4VFastSimulationModel.hh"
+#include "G4Electron.hh"
+#include "GFlashHitMaker.hh"
+#include "G4Region.hh"
+
+LowEnergyFastSimModel::LowEnergyFastSimModel(const G4String& name, G4Region* region, const edm::ParameterSet& parSet)
+    : G4VFastSimulationModel(name, region),
+      fEmin(parSet.getParameter<double>("LowEnergyGflashEcalEmin")),
+      fEmax(parSet.getParameter<double>("LowEnergyGflashEcalEmax")),
+      fRegion(region) {}
+
+G4bool LowEnergyFastSimModel::IsApplicable(const G4ParticleDefinition& particle) {
+  return &particle == G4Electron::Definition();
+}
+
+G4bool LowEnergyFastSimModel::ModelTrigger(const G4FastTrack& fastTrack) {
+  G4double energy = fastTrack.GetPrimaryTrack()->GetKineticEnergy();
+  return energy > fEmin && energy < fEmax && fRegion == fastTrack.GetEnvelope();
+}
+
+void LowEnergyFastSimModel::DoIt(const G4FastTrack& fastTrack, G4FastStep& fastStep) {
+  fastStep.KillPrimaryTrack();
+  fastStep.SetPrimaryTrackPathLength(0.0);
+  fastStep.SetTotalEnergyDeposited(fastTrack.GetPrimaryTrack()->GetKineticEnergy());
+
+  const G4double energy = fastTrack.GetPrimaryTrack()->GetKineticEnergy();
+  const G4ThreeVector pos = fastTrack.GetPrimaryTrack()->GetPosition();
+
+  G4double inPointEnergy = param.GetInPointEnergyFraction(energy) * energy;
+
+  const auto momDir = fastTrack.GetPrimaryTrack()->GetMomentumDirection();
```

Comment:
here const G4ThreeVector& seems to be better

Reply:
Yes, thanks